plugins/modules/rds_subnet_group.py

```diff
@@ -128,70 +119,67 @@ def main():
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec, required_if=required_if)
     state = module.params.get('state')
     group_name = module.params.get('name').lower()
     group_description = module.params.get('description')
-    group_subnets = module.params.get('subnets') or {}
-
-    if state == 'present':
-        for required in ['description', 'subnets']:
-            if not module.params.get(required):
-                module.fail_json(msg=str("Parameter %s required for state='present'" % required))
-    else:
-        for not_allowed in ['description', 'subnets']:
-            if module.params.get(not_allowed):
-                module.fail_json(msg=str("Parameter %s not allowed for state='absent'" % not_allowed))
-
-    # Retrieve any AWS settings from the environment.
-    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module)
-
-    if not region:
-        module.fail_json(msg=str("Either region or AWS_REGION or EC2_REGION environment variable or boto config aws_region or ec2_region must be set."))
+    group_subnets = module.params.get('subnets') or []
 
     try:
-        conn = connect_to_aws(boto.rds, region, **aws_connect_kwargs)
-    except BotoServerError as e:
-        module.fail_json(msg=e.error_message)
+        conn = module.client('rds')
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(e)
+    # Default.
+    result = create_result(False)
 
     try:
-        exists = False
-        result = create_result(False)
-
-        try:
-            matching_groups = conn.get_all_db_subnet_groups(group_name, max_records=100)
-            exists = len(matching_groups) > 0
-        except BotoServerError as e:
-            if e.error_code != 'DBSubnetGroupNotFoundFault':
-                module.fail_json(msg=e.error_message)
-
-        if state == 'absent':
-            if exists:
-                conn.delete_db_subnet_group(group_name)
-                result = create_result(True)
+        matching_groups = conn.describe_db_subnet_groups(
+            DBSubnetGroupName=group_name, MaxRecords=100).get('DBSubnetGroups')
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        # Boto3 now throws an exception instead of returning an empty list.
+        if e.response['Error']['Code'] == 'DBSubnetGroupNotFoundFault':
+            # No existing subnet, create it if needed, else we can just exit.
+            if state == 'present':
+                try:
+                    new_group = conn.create_db_subnet_group(
+                        DBSubnetGroupName=group_name, DBSubnetGroupDescription=group_description, SubnetIds=group_subnets)
+                    result = create_result(True, new_group)
+                except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+                    module.fail_json_aws(e)
+            module.exit_json(**result)
```

Comment:
```suggestion
    except is_boto3_error_code('DBSubnetGroupNotFoundFault'):
        # No existing subnet, create it if needed, else we can just exit.
        if state == 'present':
            try:
                new_group = conn.create_db_subnet_group(
                    DBSubnetGroupName=group_name, DBSubnetGroupDescription=group_description, SubnetIds=group_subnets)
                result = create_result(True, new_group)
            except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
                module.fail_json_aws(e)
            module.exit_json(**result)
    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:  # pylint: disable=duplicate-except
```

You imported is_boto3_error_code but then didn't use it.  An annoying feature of boto3 is that BotoCoreError doesn't have the return codes (it catches things like bad arguments).  is_boto3_error_code() knows that it should only be looking for ClientError **and** by only catching 'DBSubnetGroupNotFoundFault' you avoid accidentally dropping things on the floor.